### ISSUE TITLE
Need to define React as global when using CommonJS require statement

### ISSUE THREAD

**Issue by nicolashery** [jlongster/jsx-reader#2]
This might be more of a caveat than an actual issue, but thought I'd point it out. If you have:

``` javascript
var React = require('react');

var App = React.createClass({
  render: function() {
    return <div>App</div>;
  }
});

React.renderComponent(<App />, document.body);
```

Sweet.js will output (using Webpack and sweetjs-loader):

``` javascript
var React$1295 = __webpack_require__(1);

var App$1299 = React$1295.createClass({
        render: function () {
            return React.DOM.div(null, 'App');
        }
    });

React$1295.renderComponent(App$1299(null), document.body);
```

Which will throw an `Uncaught ReferenceError: React is not defined` error in the browser (because of `return React.DOM.div(null, 'App')`).

You can work around this by replacing the require statement with something like:

``` javascript
var React = window.React = require('react');
```

I don't know if there is anything else that can be done about this?

**Comment by jlongster** [comment]
Ah, very good point. I'll have to see why it's not catching the same hygiene. The read phase didn't mark hygiene, I thought. The jsx reader does some weird mangling to prefix `React.DOM` so that's probably messing things up.

**Comment by jlongster** [comment]
This should be fixed on master. I'm not publishing yet because I need to test it more thoroughly (and add automated tests). Can you test master and see if anything goes wrong?

**Comment by nicolashery** [comment]
I've tested master and it seems to be working.

Source:

``` javascript
var React = require('react');
var foo = require('./foo');

foo.bar();

var App = React.createClass({
  render: function() {
    return <div>App</div>;
  }
});

React.renderComponent(<App />, document.body);
```

Run `sjs -l jsx-loader index.js`, outputs:

``` javascript
var React$600 = require('react');
var foo$601 = require('./foo');
foo$601.bar();
var App$602 = React$600.createClass({
        render: function () {
            return React$600.DOM.div(null, 'App');
        }
    });
React$600.renderComponent(App$602(null), document.body);
```

However, I'm having issues with Webpack now (using master of https://github.com/jlongster/sweetjs-loader). The bundle comes out as:

``` javascript
/***/ function(module, exports, __webpack_require__) {

    var React$1410 = __webpack_require__(1);
    var foo$1412 = __webpack_require__(2);
    foo$1412.bar();
    var App$1414 = React.createClass({
            displayName: 'App',
            render: function () {
                return React$1410.DOM.div(null, 'App');
            }
        });
    React$1410.renderComponent(App$1414(null), document.body);

/***/ },
```

You can see that the `React.createClass` will throw an error. Is this related?

**Comment by jlongster** [comment]
It's not related, it's an issue that came up because of a fix for #1. Should be fixed on master now. Sorry if it's not; at work right not so can't property test all the changes, but will later tonight.

**Comment by nicolashery** [comment]
Thanks for the fix! Tested latest master on my little example, and all seems to be working now:

``` javascript
/***/ function(module, exports, __webpack_require__) {

    var React$1436 = __webpack_require__(1);
    var foo$1438 = __webpack_require__(2);
    foo$1438.bar();
    var App$1441 = React$1436.createClass({
            displayName: 'App',
            render: function () {
                return React$1436.DOM.div(null, 'App');
            }
        });
    React$1436.renderComponent(App$1441(null), document.body);

/***/ },
```
